### ISSUE TITLE
DevTools Support

### ISSUE THREAD

**Issue by bilenkonito** [nilproject/NiL.JS#168]
Any chance support for the devtools protocol can be added?
Right now, debugging the JS environment is quite tedious.

**Comment by deandob** [comment]
Would be helpful to know if there are any plans to enhance debugging with NiL. Its working great as the script engine for our system but debugging is a challenge. I haven't quite worked out how to effectively use the inbuilt debugging callback - ideal if we could hook it into a script IDE like codemirror or monaco.

**Comment by nilproject** [comment]
It's interesting feature. But at this moment I can not say when this will be implemented.

**Comment by deandob** [comment]
We will need to implement this sooner than later - so our team here are starting to build a basic javascript IDE using codemirror and some supporting script. Looks like it should all work, but I need to know the line number the NiL debug callback is on. Everything else we can handle ourselves. Is there any documentation for the debug callback?

**Comment by nilproject** [comment]
No, no documentation, sorry. But I made this [example](https://github.com/nilproject/NiL.JS/blob/version-2.5/Examples/8.%20Debugger%20callback/Tracing%20script%20execution.cs)

**Comment by deandob** [comment]
Thanks! Very useful and has what I need. I'd be happy to share our NiL.JS breakpoint debugger if anyone is interested.

**Comment by lafar6502** [comment]
@deandob hi, i would be interested in your project, is it available anywhere? Need code editor embedded in a web app and a debugger would be great to have.

**Comment by deandob** [comment]
@lafar6502 we have the code editor working (its easy with libraries like codemirror) but the debugger is WIP, unfortunately other features have priority at the moment. Likely mid year at this point.
